src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -622,6 +621,23 @@ bool AdaptDispatch::EraseInDisplay(const DispatchTypes::EraseType eraseType)
 
     if (success)
     {
+        // When erasing the display, every line that is erased in full should be
+        // reset to single width. When erasing to the end, this could include
+        // the current line, if the cursor is in the first column. When erasing
+        // from the beginning, though, the current line would never be included,
+        // because the cursor could never be in the rightmost column (assuming
+        // the line is double width).
+        if (eraseType == DispatchTypes::EraseType::FromBeginning)
+        {
+            const auto endRow = csbiex.dwCursorPosition.Y;
+            _pConApi->PrivateResetLineRenditionRange(csbiex.srWindow.Top, endRow);
```

Comment:
Why reset the line rendition separately here instead of inside of `DoSrvPrivateFillRegion` or even further down like in every consumer of `screenInfo.Write`/`TextBuffer::WriteLine` much like how line wraps are added/removed/left alone in certain common cases like rectangular fills?

Reply:
I did consider doing something like that, but not all rectangular fills are meant to reset the line rendition. For example you could clear a double-width line with `EL` or `ECH`, and that would result in the full width being filled, but shouldn't reset the rendition.

Reply:
Fair enough.